lib/grid.rb

```diff
@@ -17,9 +17,23 @@ def get_value(coordinate)
     structure[coordinate]
   end
 
+  def get_computer_choice
+    get_random_key
+  end
+
   private
 
   def grid_structure
     {"A1"=> ' ', "A2"=> ' ', "A3"=> ' ', "B1"=> ' ', "B2"=> ' ', "B3"=> ' ', "C1"=> ' ', "C2"=> ' ', "C3"=> ' '}
   end
+
+  def get_random_key
```

Comment:
This is pretty complex. Also it breaks the rules about 5 lines per method! So two thoughts:
1. Break this apart to see what the method names reveal
2. Think about whether the data structure is right for your full needs. You started off thinking of exposing this hash structure and allowing people to access values by key directly. Now that it is encapsulated here, and also that you need to access it differently, is there a better structure for the data?

Reply:
I've got a different strategy now. Let me know what you think.